test/new-e2e/tests/containers/base_test.go

```diff
@@ -11,40 +11,96 @@ import (
 	"strings"
 	"time"
 
+	"github.com/samber/lo"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/suite"
 
+	"github.com/DataDog/agent-payload/v5/gogen"
 	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
 	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
 )
 
 type baseSuite struct {
 	suite.Suite
 
+	startTime  time.Time
+	endTime    time.Time
 	Fakeintake *fakeintake.Client
 }
 
-func (suite *baseSuite) testMetric(metricName string, filterTags []string, expectedTags []*regexp.Regexp) {
-	suite.Run(fmt.Sprintf("%s{%s}", metricName, strings.Join(filterTags, ",")), func() {
+func (suite *baseSuite) SetupSuite() {
+	suite.startTime = time.Now()
+}
+
+func (suite *baseSuite) TearDownSuite() {
+	suite.endTime = time.Now()
+}
+
+type testMetricArgs struct {
+	filter testMetricFilterArgs
+	expect testMetricExpectArgs
+}
+
+type testMetricFilterArgs struct {
+	name string
+	tags []string
+}
+
+type testMetricExpectArgs struct {
+	tags  *[]string
+	value *testMetricExpectValueArgs
+}
+
+type testMetricExpectValueArgs struct {
+	min float64
+	max float64
+}
+
+func (suite *baseSuite) testMetric(args *testMetricArgs) {
+	//metricName string, filterTags []string, expectedTags []*regexp.Regexp) {
```

Comment:
Delete?

Reply:
🤦🏻 Of course !
Thanks for catching it !!